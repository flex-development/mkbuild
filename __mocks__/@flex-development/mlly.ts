/**
 * @file Mocks - @flex-development/mlly
 * @module mocks/flex-development/mlly
 * @see https://github.com/flex-development/mlly
 */

/**
 * [`@flex-development/mlly`][1] module type.
 *
 * [1]: https://github.com/flex-development/mlly
 */
type Actual = typeof import('@flex-development/mlly')

/**
 * `@flex-development/mlly` module.
 *
 * @const {Actual} actual
 */
const actual: Actual = await vi.importActual<Actual>('@flex-development/mlly')

export const CONDITIONS = actual.CONDITIONS
export const RESOLVE_EXTENSIONS = actual.RESOLVE_EXTENSIONS
export const fillModules = vi.fn(actual.fillModules)
export const findStaticImports = vi.fn(actual.findStaticImports)
export const getSource = vi.fn(actual.getSource)
export const readPackageJson = vi.fn(actual.readPackageJson)
export const resolveModule = vi.fn(actual.resolveModule)
export const resolveModules = vi.fn(actual.resolveModules)
export const toDataURL = vi.fn(actual.toDataURL)
export const toURL = vi.fn(actual.toURL)
