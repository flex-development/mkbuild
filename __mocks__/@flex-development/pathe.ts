/**
 * @file Mocks - @flex-development/pathe
 * @module mocks/flex-development/pathe
 * @see https://github.com/flex-development/pathe
 */

/**
 * [`@flex-development/pathe`][1] module type.
 *
 * [1]: https://github.com/flex-development/pathe
 */
type Actual = typeof import('@flex-development/pathe')

/**
 * `@flex-development/pathe` module.
 *
 * @const {Actual} actual
 */
const actual: Actual = await vi.importActual<Actual>('@flex-development/pathe')

export const basename = vi.fn(actual.basename)
export const dirname = vi.fn(actual.dirname)
export const extname = vi.fn(actual.extname)
export const format = vi.fn(actual.format)
export const join = vi.fn(actual.join)
export const parse = vi.fn(actual.parse)
export const relative = vi.fn(actual.relative)
export const resolve = vi.fn(actual.resolve)

export default {
  basename,
  dirname,
  extname,
  format,
  join,
  parse,
  relative,
  resolve
}
