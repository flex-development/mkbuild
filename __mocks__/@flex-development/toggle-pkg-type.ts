/**
 * @file Mocks - @flex-development/toggle-pkg-type
 * @module mocks/flex-development/toggle-pkg-type
 * @see https://github.com/flex-development/toggle-pkg-type
 */

export default vi.fn()
