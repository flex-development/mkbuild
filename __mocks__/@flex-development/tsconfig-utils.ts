/**
 * @file Mocks - @flex-development/tsconfig-utils
 * @module mocks/flex-development/tsconfig-utils
 * @see https://github.com/flex-development/tsconfig-utils
 */

/**
 * [`@flex-development/tsconfig-utils`][1] module type.
 *
 * [1]: https://github.com/flex-development/tsconfig-utils
 */
type Actual = typeof import('@flex-development/tsconfig-utils')

/**
 * `@flex-development/tsconfig-utils` module.
 *
 * @const {Actual} actual
 */
const actual: Actual = await vi.importActual<Actual>(
  '@flex-development/tsconfig-utils'
)

export const ScriptTarget = actual.ScriptTarget
export const normalizeCompilerOptions = vi.fn(actual.normalizeCompilerOptions)
export const normalizeModuleResolution = vi.fn(actual.normalizeModuleResolution)
export const resolvePaths = vi.fn(actual.resolvePaths)
