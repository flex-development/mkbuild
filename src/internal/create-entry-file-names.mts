/**
 * @file Internal - createEntryFileNames
 * @module mkbuild/internal/createEntryFileNames
 */

import type { EntryFileNamesFn, Task } from '@flex-development/mkbuild'
import { isNIL, template } from '@flex-development/tutils'
import { ok } from 'devlop'
import type { PreRenderedChunk } from 'rollup'

/**
 * Create a function that returns a pattern to use for naming chunks created
 * from entry points.
 *
 * @see {@linkcode EntryFileNamesFn}
 * @see {@linkcode Task}
 *
 * @internal
 *
 * @this {void}
 *
 * @param {Pick<Task, 'entryFileNames' | 'ext'>} task
 *  Build task
 * @return {EntryFileNamesFn}
 *  Entry point chunk file name generator
 */
function createEntryFileNames(
  this: void,
  task: Pick<Task, 'entryFileNames' | 'ext'>
): EntryFileNamesFn {
  return entryFileNames

  /**
   * @this {void}
   *
   * @param {PreRenderedChunk} chunk
   *  Pre-rendered entry chunk
   * @return {string}
   *  Pattern to use for naming `chunk`
   */
  function entryFileNames(this: void, chunk: PreRenderedChunk): string {
    ok(task, 'expected `task`')
    ok(!isNIL(task.entryFileNames), 'expected `task.entryFileNames`')
    ok(!isNIL(task.ext), 'expected `task.ext`')

    /**
     * Pattern to use for naming chunks created from entry points.
     *
     * @const {string} entryFileNames
     */
    const entryFileNames: string = typeof task.entryFileNames === 'string'
      ? task.entryFileNames
      : task.entryFileNames(chunk)

    return template(entryFileNames, { extname: task.ext })
  }
}

export default createEntryFileNames
