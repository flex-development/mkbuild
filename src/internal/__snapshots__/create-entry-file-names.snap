// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`unit:internal/createEntryFileNames > entryFileNames > should return pattern to use for naming \`chunk\` (0) 1`] = `"[name].mjs"`;

exports[`unit:internal/createEntryFileNames > entryFileNames > should return pattern to use for naming \`chunk\` (1) 1`] = `"make.mjs"`;
